### PR TITLE
Address PoA audit code quality issues

### PR DESCRIPTION
**Pull request by sergeyshemyakov**
Resolve https://github.com/LiskHQ/lips/issues/481

### CODE REVIEW

**Review by vjaiman — APPROVED** [review]
Looks good to me.

File: proposals/lip-0047.md [hunk 1]
```diff
@@ -763,19 +747,16 @@ def afterTransactionsExecute(b: Block) -> None:
         roundStartHeight = chainProperties.roundEndHeight - previousLengthValidators + 1
         randomSeed = Random.getRandomBytes(roundStartHeight, previousLengthValidators)
 
-        addresses = []
-        for i in range(length(snapshotStore(0).validators)):
-            addresses[i] =  snapshotStore(0).validators[i].address
-
-        nextValidatorAddresses  = shuffleValidatorsList(addresses, randomSeed)
-        Validators.setValidatorParams(snapshotStore(0).threshold, snapshotStore(0).threshold, nextValidatorAddresses)
+        nextValidators  = PoS.shuffleValidatorsList(snapshotStore(0).validators, randomSeed)
```
Comment (gkoumout) [inline proposals/lip-0047.md:750]: I find this an undesirable dependency: the idea is that chains should use PoA OR PoS, so it does not make much sense IMO to call a PoS endpoint from PoA module. 

Perhaps the best would be if we move the `shuffleValidatorsList` to random module, but I need to think a bit about it.

**Review by gkoumout — CHANGES_REQUESTED** [review]

**Review by mitsuaki-u — APPROVED** [review]

**Review by gkoumout — CHANGES_REQUESTED** [review]

File: proposals/lip-0047.md [hunk 2]
```diff
@@ -572,22 +572,7 @@ validatorSignatureMessageSchema = {
...
-    for address in validatorsAddresses:
-        roundHash[address] = SHA-256(randomSeed + address)
-
-    # Reorder the validator list.
-    shuffledValidatorAddresses = sort validatorsAddresses where address1 < address2 if (roundHash[address1] < roundHash[address2])
-                                 or ((roundHash[address1] == roundHash[address2]) and address1 < address2)         
-
-    return shuffledValidatorAddresses
-```
+The function reorders and returns the list of validators as specified in [LIP 0003][lip-0003#specs]. It must be implemented exactly as the `shuffleValidatorsList` function in the [PoS module](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#shufflevalidatorslist).
```
Comment (janhack) [inline proposals/lip-0047.md:575]:
```suggestion
The function reorders and returns the list of validators. It must be implemented exactly as the `shuffleValidatorsList` function in the [PoS module](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#shufflevalidatorslist).
```

Comment (janhack) [inline proposals/lip-0047.md:575]:
```suggestion
The function reorders and returns the list of validators. It must be implemented exactly as the `shuffleValidatorsList` function in the [PoS module](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0057.md#shufflevalidatorslist).
```

Comment (janhack) [reply proposals/lip-0047.md:575]: I think the historic reference to LIP 0003 is not important here and rather confusing (there should be one source of truth for the specifications).

**Review by janhack — CHANGES_REQUESTED** [review]
Looks good, I just left one suggestion.

**Review by gkoumout — APPROVED** [review]

**Review by janhack — APPROVED** [review]
Editorial approval

**Pull request merged**